lib/kdl/include/kdl/result.h

```diff
@@ -446,6 +545,62 @@ namespace kdl {
                 return visitor();
             }
         }
+        
+        /**
+         * Applies the given function to given result and returns a new result with the result type of
```

Comment:
This first paragraph should just be deleted IMO, it's imprecise and confusing more than anything. The next paragraph (If the given result is a success... If the given result contains an error...) is clear and about as concise of an explanation as we can give I think.

Reply:
I'll make these changes in a later PR: https://github.com/kduske/TrenchBroom/issues/2983#issuecomment-658147940